src/sb_laserscan_processing/src/ConeIdentification.cpp

```diff
@@ -0,0 +1,293 @@
+/**
+ * Created by William Gu on 24/03/18
+ * Implementation of Cone Identification methods for identifying cones from a laser msg
+ */
+
+#include <ConeIdentification.h>
+
+
+std::vector<mapping_igvc::ConeObstacle> ConeIdentification::identifyCones(const sensor_msgs::LaserScan &laser_msg, double dist_tol, double radius_exp, double radius_tol, int line_point_dist, double ang_threshold){
```

Comment:
This function is pretty long (>100), I think it would be much more readable if we could break it up into functions.